packages/canon/src/components/Header/Header.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { HeaderProps } from './types';
+import { HeaderToolbar } from './HeaderToolbar';
+import { HeaderTabs } from './HeaderTabs';
+import { HeaderSubNav } from './HeaderSubNav';
+
+/**
+ * A component that renders a toolbar.
+ *
+ * @public
+ */
+export const Header = (props: HeaderProps) => {
```

Comment:
Having a bit of a hard time wrapping my head around what the scope of this component is. It seems to go a bit beyond just the header and more towards implementing the full page with navigation in a fairly opinionated way. I have a bit of a feeling we may want to split things up a little bit. Especially curious about the purpose of the SubNav

Reply:
Tricky bit is that often the page header and the page content are not part of the same extension or even plugin, so it's less about naming and more about that the component may be crossing what is often a plugin boundary. I'm thinking at least the subnav might be good to leave as a separate component.

Tabs are probably fine, they tend to be owned by the page but sometimes populated by other plugins. The main bit there is to make sure we can tie them into react-router nicely